### ISSUE TITLE
require-jsdoc: constructor overload signatures incorrectly require JSDoc

### ISSUE THREAD

**Issue by HolgerJeromin** [gajus/eslint-plugin-jsdoc#1727]
I found a similar issue like https://github.com/gajus/eslint-plugin-jsdoc/issues/1688

cc @brettz9 

## Expected behavior
Because `skipInterveningOverloadedDeclarations` is `true` this should be fine IMO.

## Actual behavior

`error  Missing JSDoc comment  jsdoc/require-jsdoc`

## ESLint Config

eslint.config.mjs
```js
'jsdoc/require-jsdoc': [
    'error',
    {
        publicOnly: true,
        require: {
            MethodDefinition: true,
        },
    },
]
```

## ESLint sample

```ts
/** Foo here */
export class Foo {
    /** with string */
    constructor(arg1: string);
    /** with number */
    constructor(arg1: number);
  
    constructor(arg1: string | number) {   // <--- Missing JSDoc comment  jsdoc/require-jsdoc
        // implementation
    }
}
```

## Environment

- Node version: 24
- ESLint version 10
- `eslint-plugin-jsdoc` version: 63.0.13 (which includes https://github.com/gajus/eslint-plugin-jsdoc/pull/1718 )

Thanks a lot for fixing these issues. Deleting many `eslint-disable-next-line` feels very good :)

**Comment by HolgerJeromin** [comment]
wow. That was quick
Thanks a lot!
